src/memory.js

```diff
@@ -0,0 +1,42 @@
+
+import loadROM from './utils/loadROM.js';
+
+class Memory {
+
+  constructor(rom) {
+
+    if (!rom) {
+      throw new Error('requires ROM');
+    }
+
+    this.data = loadROM(rom);
+  }
+
+  readROM(value) {
+    let romValue;
+    if (typeof value != 'number') {
```

Comment:
Could you help me understand the case when `value` is not a number?

Reply:
if it's undefined... also if I just did if (!value), 0 would trigger it too...

Reply:
No judgment here. I was trying to understand the reasoning for the `number` check because it changes the signature of the method. Instead of returning a Number or null, it returns a very large TypedArray

```
// Number
readROM(0x1234) === Number;
// Not number
readROM(null) === readRom(undefined) === void 0;
// Non-number, returns entire ROM
readROM('booger') === TypedArray;
```

Reply:
Oh right, in order to get the whole ROM, you just call and don't give it an address. There was some reason around that too.